cmake/install.cmake

```diff
@@ -6,6 +6,9 @@ foreach(_library
   libprotoc)
   set_property(TARGET ${_library}
     PROPERTY INTERFACE_INCLUDE_DIRECTORIES
+    $<BUILD_INTERFACE:${protobuf_BINARY_DIR}>
```

Comment:
Given the structure of the cmake files, it seems inappropriate to add these here. 
If a special install.cmake file exists, it implies that it should contain exclusively logic related to installation. Specifying BUILD_INTERFACE rules should happen in the <library>.cmake file.